datafusion/src/optimizer/utils.rs

```diff
@@ -195,11 +195,21 @@ pub fn from_plan(
             schema: schema.clone(),
             alias: alias.clone(),
         }),
+        LogicalPlan::Analyze {
+            verbose, schema, ..
+        } => {
+            assert!(expr.is_empty());
```

Comment:
Here is the actual bug fix: the code that builds up LogicalPlans from optimized inputs ignored the inputs for Analyze. I have a follow on PR to try and make this kind of thing harder to introduce in the future:  https://github.com/apache/arrow-datafusion/pull/930